### PR TITLE
Various code and test fixes

### LINKED ISSUE #140: Stylesheet URLs broken when using ugly permalinks
**Describe the bug**
Noticed today while working on some tests.

**To Reproduce**
1. Disable pretty permalinks
1. Open sitemap or sitemap index
1. Notice that stylesheet URL is pretty, not ugly

### PR DESCRIPTION
**Pull request by swissspidy**
* Fix stylesheet URLs when pretty permalinks are disabled
* Add specific tests for registry
* Split some tests into separate files
* Add a couple of new tests

Fixes #140.

### CODE REVIEW

File: inc/class-core-sitemaps.php [hunk 1]
```diff
@@ -120,9 +120,9 @@ public function register_rewrites() {
 		add_rewrite_rule( '^wp-sitemap\.xml$', 'index.php?sitemap=index', 'top' );
 
 		// Register rewrites for the XSL stylesheet.
-		add_rewrite_tag( '%stylesheet%', '([^?]+)' );
-		add_rewrite_rule( '^wp-sitemap\.xsl$', 'index.php?stylesheet=xsl', 'top' );
-		add_rewrite_rule( '^wp-sitemap-index\.xsl$', 'index.php?stylesheet=index', 'top' );
+		add_rewrite_tag( '%sitemap-stylesheet%', '([^?]+)' );
```
Comment (swissspidy) [inline inc/class-core-sitemaps.php:123]: Just `stylesheet` was too ambiguous.

File: phpunit.xml.dist [hunk 2]
```diff
@@ -15,8 +15,7 @@
 
 	<testsuites>
 		<testsuite name="plugins">
-			<directory prefix="class-test-" suffix=".php">./tests/phpunit/</directory>
+			<directory suffix=".php">./tests/phpunit/</directory>
```
Comment (swissspidy) [inline phpunit.xml.dist:18]: This is more in line with WordPress core.

File: tests/phpunit/sitemaps-renderer.php [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+class Test_Core_Sitemaps_Renderer extends WP_UnitTestCase {
+	public function test_get_sitemap_stylesheet_url() {
+		$sitemap_renderer = new Core_Sitemaps_Renderer();
+		$stylesheet_url = $sitemap_renderer->get_sitemap_stylesheet_url();
+
+		$this->assertStringEndsWith( '/?sitemap-stylesheet=xsl', $stylesheet_url );
+	}
+
+	public function test_get_sitemap_stylesheet_url_pretty_permalinks_pretty_permalinks() {
```
Comment (joemcgill) [inline tests/phpunit/sitemaps-renderer.php]: `pretty_permalinks` x 2

File: tests/phpunit/sitemaps-renderer.php [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+	}
+
+	public function test_get_sitemap_index_stylesheet_url() {
+		$sitemap_renderer = new Core_Sitemaps_Renderer();
+		$stylesheet_url = $sitemap_renderer->get_sitemap_index_stylesheet_url();
+
+		$this->assertStringEndsWith( '/?sitemap-stylesheet=index', $stylesheet_url );
+	}
+
+	public function test_get_sitemap_index_stylesheet_url_pretty_permalinks_pretty_permalinks() {
```
Comment (joemcgill) [inline tests/phpunit/sitemaps-renderer.php]: `pretty_permalinks` x 2

**Review by joemcgill — CHANGES_REQUESTED** [review]
Really nice break up of the tests into individual classes. Makes it much easier to read through 🥇 

Left a couple of notes about the test method names for the new stylesheet tests being redundant, otherwise it looks good.

File: tests/phpunit/sitemaps-renderer.php (hunk 3 above)
Comment (swissspidy) [reply tests/phpunit/sitemaps-renderer.php]: They're especially pretty! 😄

Comment (swissspidy) [reply tests/phpunit/sitemaps-renderer.php]: Fixed now!

File: tests/phpunit/sitemaps-renderer.php (hunk 4 above)
Comment (swissspidy) [reply tests/phpunit/sitemaps-renderer.php]: Fixed now!

**Review by joemcgill — APPROVED** [review]
👏 Less pretty.

**Pull request merged**